### PR TITLE
Fall back to "/" when home directory doesn't exist for `tsh ssh`

### LINKED ISSUE #2834: Users without home directory will not be able to log-in
**What happened**:
When I try to connect to a remote node I some times get the following error.
```
ERRO [NODE]      Unable to run shell command (bash): fork/exec /bin/bash: no such file or directory id:30 local:127.0.0.1:3022 login:<username> remote:xxx.xxx.xxx.xxx:58860 teleportUser:ioannis srv/sess.go:605
ERRO [NODE]      fork/exec /bin/bash: no such file or directory regular/sshserver.go:1230
```
**What you expected to happen**:
Login to the node successfully.

**How to reproduce it (as minimally and precisely as possible)**:
Create a user with a home directory pointing to a non existing location. Try to login to that node with that user. After creating the home directory the problem goes away.

**Environment**:
- Teleport version (use `teleport version`):
Teleport v4.0.0-rc.3 git:v4.0.0-rc.3-2-g1544d40 go1.10.3
- Tsh version (use `tsh version`):
Teleport v4.0.0-rc.3 git:v4.0.0-rc.3-2-g1544d40 go1.10.3
- OS (e.g. from /etc/os-release):

**Browser environment**

- Browser Version (for UI-related issues):
- Install tools:
- Others:

**Relevant Debug Logs If Applicable**

- tsh --debug
- teleport --debug

### PR DESCRIPTION
**Pull request by atburke**
This PR changes `tsh ssh`'s behavior when a user's `$HOME` points to a non-existent directory. Previously, this caused the command to fail; now, the user's home directory is set to `/` (this is the same behavior as OpenSSH). 

Fixes #2834.

### CODE REVIEW

File: lib/srv/reexec.go [hunk 1]
```diff
@@ -423,11 +423,17 @@ func buildCommand(c *ExecCommand, tty *os.File, pty *os.File, pamEnvironment []s
 		cmd.Args = []string{shellPath, "-c", c.Command}
 	}
 
+	// Ensure that the user has a home directory.
+	homeDir := "/"
```
Comment (russjones) [inline lib/srv/reexec.go:427]: `os.PathSeparator`?

**Review by russjones — COMMENTED** [review]
Can you add test coverage?

**Comment by zmb3** [comment]
Note that assuming `os.PathSeparator` is a valid directory only works on Linux.

This is probably okay for now, but eventually we'll be running teleport as a Windows service, so could we at least add a `// TODO` here for Windows support so it's one less random bug to chase down if we want this to work on Windows some day?

**Comment by russjones** [comment]
@atburke Can you backport this to `branch/v7` and `branch/v8`.

**Review by zmb3 — APPROVED** [review]

**Comment by atburke** [comment]
@gabrielcorado Can you review this when you have a moment?

**Review by gabrielcorado — APPROVED** [review]
Sorry for not getting it reviewed sooner.

**Comment by 2022liuws** [comment]
disconnected
Failed to launch: fork/exec /proc/1544650/fd/8: permission denied.
Process exited with status 255

if without home directory will

**Pull request merged**